Makefile

```diff
@@ -190,27 +190,26 @@ unit-sup: build-launcher-for-supervisor-tests
 .PHONY: build-launcher-for-supervisor-tests
 
 # Lints we need to work through and decide as a team whether to allow or fix
-UNEXAMINED_LINTS = clippy::cyclomatic_complexity \
-                   clippy::large_enum_variant \
-                   clippy::len_without_is_empty \
-                   clippy::module_inception \
-                   clippy::needless_pass_by_value \
-                   clippy::needless_return \
-                   clippy::new_ret_no_self \
-                   clippy::new_without_default \
-                   clippy::new_without_default_derive \
-                   clippy::question_mark \
-                   clippy::redundant_field_names \
-                   clippy::too_many_arguments \
-                   clippy::trivially_copy_pass_by_ref \
-                   clippy::wrong_self_convention \
-                   renamed_and_removed_lints
+UNEXAMINED_LINTS =
 
 # Lints we disagree with and choose to keep in our code with no warning
-ALLOWED_LINTS =
+ALLOWED_LINTS = clippy::module_inception \
+                clippy::needless_return \
+                clippy::new_ret_no_self \
+                clippy::new_without_default \
+                clippy::new_without_default_derive \
+                clippy::redundant_field_names
 
 # Known failing lints we want to receive warnings for, but not fail the build
-LINTS_TO_FIX =
+LINTS_TO_FIX = clippy::cyclomatic_complexity \
+               clippy::large_enum_variant \
+               clippy::len_without_is_empty \
+               clippy::needless_pass_by_value \
+               clippy::question_mark \
+               clippy::too_many_arguments \
```

Comment:
https://rust-lang-nursery.github.io/rust-clippy/master/index.html#too_many_arguments

I'm not totally opposed to `#[allow(…)]`'ing some cases, but I think this is generally good practice since this makes things pretty hard to understand at the call sites.

[too_many_arguments.txt](https://github.com/habitat-sh/habitat/files/2814087/too_many_arguments.txt)

Reply:
Wow, some of those argument lists are bonkers.  Let's do this.